File: kevoree-core/org.kevoree.api/src/main/java/org/kevoree/api/adaptation/Step.java
package org.kevoree.api.adaptation;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by duke on 8/6/14.
 */

public class Step {
    private List<AdaptationPrimitive> adaptations = new ArrayList<AdaptationPrimitive>();
    private Step nextStep;
    private AdaptationType adaptationType;

    public List<AdaptationPrimitive> getAdaptations() {
        return adaptations;
    }

    public void setAdaptations(List<AdaptationPrimitive> adaptations) {
        this.adaptations = adaptations;
    }

    public Step getNextStep() {
        return nextStep;
    }

    public void setNextStep(Step nextStep) {
        this.nextStep = nextStep;
    }

    public Step next(Step next) {
        this.setNextStep(next);
        return next;
    }

    public AdaptationType getAdaptationType() {
        return adaptationType;
    }

    public void setAdaptationType(AdaptationType adaptationType) {
        this.adaptationType = adaptationType;
    }

}

File: kevoree-core/org.kevoree.api/src/main/java/org/kevoree/api/adaptation/Step.kt
package org.kevoree.api.adaptation

import java.util.ArrayList

/**
 * Created by duke on 8/6/14.
 */

open class Step {
    var adaptations : MutableList<AdaptationPrimitive> = ArrayList<AdaptationPrimitive>()
    var nextStep : Step? = null
}

class ParallelStep : Step() {

}

class SequentialStep : Step() {

}